src/cairoUtilFuncGen/memory/arrayLiteral.ts

```diff
@@ -117,44 +113,53 @@ export class MemoryArrayLiteralGen extends StringIndexedFuncGen {
     return funcDef;
   }
 
-  private getOrCreate(type: TypeNode, size: number, dynamic: boolean): GeneratedFunctionInfo {
-    const elementCairoType = CairoType.fromSol(type, this.ast);
+  private getOrCreate(elementType: TypeNode, arraySize: number): GeneratedFunctionInfo {
+    const elementCairoType = CairoType.fromSol(elementType, this.ast);
+    const dynamic = isDynamicArray(elementType) || elementType instanceof StringLiteralType;
+
     const funcName = `wm${this.generatedFunctionsDef.size}_${dynamic ? 'dynamic' : 'static'}_array`;
+    const argString = mapRange(arraySize, (n) => `e${n}: ${elementCairoType.toString()}`).join(
+      ', ',
+    );
 
-    const argString = mapRange(size, (n) => `e${n}: ${elementCairoType.toString()}`).join(', ');
+    const alloc_len: number = dynamic
+      ? arraySize * elementCairoType.width + 1
+      : arraySize * elementCairoType.width;
 
-    // If it's dynamic we need to include the length at the start
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const alloc_len = dynamic ? size * elementCairoType.width + 2 : size * elementCairoType.width;
+    const funcCall =
+      elementCairoType instanceof CairoFelt
+        ? this.requireImport(...WM_WRITE)
+        : this.requireImport(...WM_STORE);
     const writes = [
-      ...(dynamic ? [`wm_write_256{warp_memory=warp_memory}(start, ${uint256(size)});`] : []),
-      ...mapRange(size, (n) => elementCairoType.serialiseMembers(`e${n}`))
+      dynamic ? `warp_memory.write(start, ${arraySize});` : '',
```

Comment:
Do you mean the `writes`? I prefer it outside, inside it makes it hard to understand what happens in the generated function

Reply:
I meant only the write of dynamic array's length. I agree writing elements outside is better. The length though is just a single write so I think it will be more visible if it is there